clang/docs/TypeSanitizer.rst

```diff
@@ -0,0 +1,153 @@
+================
+TypeSanitizer
+================
+
+.. contents::
+   :local:
+
+Introduction
+============
+
+TypeSanitizer is a detector for strict type aliasing violations. It consists of a compiler
+instrumentation module and a run-time library. The tool detects violations such as the use 
+of an illegally cast pointer, or misuse of a union.
+
+The violations TypeSanitizer catches may cause the compiler to emit incorrect code.
+
+Typical slowdown introduced by TypeSanitizer is about **4x** [[CHECK THIS]]. Typical memory overhead introduced by TypeSanitizer is about **9x**. 
```

Comment:
Might be good to just say that TypeSanitizer is still experimental and currently can have a very large runtime, memory and code size overhead :)

Reply:
It also seems to have a sizable compile-time overhead.